### ISSUE TITLE
Allow JetMenu to switch to the mobile menu by breakpoints as well as device.

### ISSUE THREAD

**Issue by robelia** [CrocoBlock/suggestions#2252]
Currently, JetMenu only changes from the desktop menu to the mobile menu when it detects a mobile device. It would be great to have the menu to mobile based on breakpoints as well. Only using device detection leaves out any users that might be viewing the site in a smaller window on their desktop device.

**Comment by uborkh** [comment]
Agree with you

**Comment by bowmandesign** [comment]
I agree, this needs addressed.

The requirement and better solution would be to show the mobile menu if the user minimizes their desktop browser to the mobile breakpoint, as well as allow Crocoblock customers to have mobile and tablet share the mobile/hamburger menu, while having a unique desktop menu, or even have a unique and standalone mobile, tablet, and desktop menu that responds responsively on custom breakpoints, regardless of which device the user is on. (Like Elementor)

My challenge is that my clients struggle with how JetMenu forces the desktop view when they resize their desktop browser to the tablet or mobile responsive breakpoint, which happens a lot. The result is that the larger desktop menu uncomfortably wraps in the smaller browser window, even though the mobile breakpoint was reached on desktop.

Also, not having a tablet view menu option in JetMenu is a problem. JetMenu forces tablet views to inherit the mobile or desktop view globally. 

JetMenu Device View does not allow a user to have the mobile view on mobile and tablet, while having a different view on desktop. The pivot breakpoints are limiting, only allowing:

{ desktop on all devices }
or 
{ mobile on all devices }
 or 
{ a different menu on mobile and desktop only }
missing
{  the ability to inherit the mobile view on tablet, while having a unique desktop view and better yet, a unique tablet view. }

This important missing functionality forces customers to make a work around, using multiple menus and leveraging the responsive settings in Elementor to show/hide menus correctly to accommodate mobile, tablet, and desktop browsing. 

Beware, having multiple menus drives Google to incorrectly index your site and favor menu ID names rather than menu label names, example “mega-item-5110” can be displayed rather than correctly displaying the menu navigation label. 

PS
The help text in JetMenu/Content/General has a misspelling. 

Choose “witch” menu view you want to display
s/b
Choose “which” menu view you want to display

**Comment by KingGrizzly** [comment]
+1 Would be nice if the menu had mobile, tablet, and desktop options just like Elementor. Choose which menu shows for each of those.

**Comment by srack835** [comment]
+1 Yes! Many users will shrink the window to essentially a mobile size and I would like the menu to respond at breakpoints, just like Elementor.

Also, this makes it really difficult to try to preview the Jet Menu mobile menu in the Elementor editor.

**Comment by UpDigital-Studio** [comment]
+1 for this.

**Comment by jurijboni** [comment]
after a year it still hasn't been resolved

**Comment by DynamoProd** [comment]
+1 We definitly need this

**Comment by fabio323ti** [comment]
With latest update, mobile render layout (slide out) not working on tablet breakpoint 1024, menu is displayed in dropdown layout mode. everytime ther's a bug !

**Comment by UpDigital-Studio** [comment]
Would love to see a response from @Crocoblock on this one. This is a major flaw in the plugin and forces users to create workarounds to make sure the menu appears correctly on a browser.

It's critical that this is fixed.

Are there any plans to resolve this issue in upcoming updates @Crocoblock?

**Comment by fabio323ti** [comment]
i finally used two menu "inline" one mega and one hamburger, each of them with responsive settings (hide the first on desktop and tabled, and show second on them), i know its not a lunar exploit but it works..

**Comment by tony-keller** [comment]
+1

**Comment by Smalyleinchen** [comment]
+1

**Comment by David-Heigl** [comment]
+1

**Comment by proactivemel** [comment]
+1 

Very glitchy when window resizing is in play. Any advice on improving it, much appreciated.

**Comment by David-Heigl** [comment]
+1

**Comment by tropicchaotic** [comment]
+1

**Comment by SamuelAshbrook** [comment]
+1

**Comment by GALASIO** [comment]
+1

**Comment by jodamo5** [comment]
+1

**Comment by wpfixy** [comment]
+1

**Comment by antoinelbch** [comment]
+1

**Comment by MaximPro** [comment]
???

**Comment by e-cubed** [comment]
Any updates on this one?

**Comment by jonneebee** [comment]
Agreed, this is silly that we cant just rely on the elementor breakpoints at this point.  Or even more simply, set a width we'd like it to display mobile on.

**Comment by jason3by2** [comment]
How is this still not a thing yet?

**Comment by Fadirko** [comment]
+1 @Crocoblock

**Comment by mbaierl** [comment]
Any alternatives you can recommend? The Elementor Mega Menü works way better on Desktop, but is a pain on mobile...

**Comment by apatsidis123** [comment]
+1 end of 2024 still breaks by device type, we cant the responsiveness with the tools that emulate mobile devices from a pc, it recognizes them as desktops and we see the desktop version that is not a good idea in general!
